### PR TITLE
Refactor(shadow mouse) - externalize as visualizer

### PR DESCRIPTION
**Pull request by yuvalbl**
related to #768

### CODE REVIEW

File: src/app/app/app-effects/effects/map.app.effects.ts [hunk 1]
```diff
@@ -82,9 +75,11 @@ import { overlaysStateSelector } from '@ansyn/overlays/reducers/overlays.reducer
 import { IMapFacadeConfig } from '@ansyn/map-facade/models/map-config.model';
 import { mapFacadeConfig } from '@ansyn/map-facade/models/map-facade.config';
 import { getPolygonByPointAndRadius } from '@ansyn/core/utils/geo';
+import { StartMouseShadow } from '@ansyn/menu-items/tools/actions/tools.actions';
 import 'rxjs/add/observable/fromPromise';
 
 import { ILayerState, layersStateSelector } from '@ansyn/menu-items/layers-manager/reducers/layers.reducer';
+import { dispatch } from 'd3-dispatch';
```
Comment (yairtawil) [inline src/app/app/app-effects/effects/map.app.effects.ts]: web storm

File: src/app/app/app-effects/effects/map.app.effects.ts [hunk 2]
```diff
@@ -403,43 +376,70 @@ export class MapAppEffects {
...
 		.ofType(MapActionTypes.ADD_MAP_INSTANCE, MapActionTypes.MAP_INSTANCE_CHANGED_ACTION)
 		.withLatestFrom(this.store$.select(casesStateSelector), this.store$.select(statusBarStateSelector))
-		.filter(([action, casesState, statusBarState]: [any, ICasesState, IStatusBarState]) => statusBarState.flags.get(statusBarFlagsItems.pinPointIndicator) || statusBarState.flags.get(statusBarFlagsItems.pinPointSearch))
-		.map(([action, casesState, statusBarState]: [any, ICasesState, IStatusBarState]) => {
-			const communicatorHandler = this.imageryCommunicatorService.provide(action.payload.currentCommunicatorId);
+		.filter(([action, casesState, statusBarState]: [any, ICasesState, IStatusBarState]) =>
+			statusBarState.flags.get(statusBarFlagsItems.pinPointIndicator))
+		.map(([action, casesState]: [any, ICasesState]) => {
+			const point = getPointByPolygon(casesState.selectedCase.state.region);
+			return new DrawPinPointAction(point.coordinates);
```
Comment (yairtawil) [inline src/app/app/app-effects/effects/map.app.effects.ts]: Great job!

File: src/app/app/app-effects/effects/map.app.effects.ts [hunk 3]
```diff
@@ -403,43 +376,70 @@ export class MapAppEffects {
...
+	 * @dependencies cases, statusBar
+	 * @filter shadowMouse flag on
+	 * @actions StartMouseShadow
+	 */
+	@Effect()
+	onAddCommunicatorShowShadowMouse$: Observable<any>  = this.actions$
+		.ofType(MapActionTypes.ADD_MAP_INSTANCE, MapActionTypes.MAP_INSTANCE_CHANGED_ACTION,  MapActionTypes.STORE.SET_MAPS_DATA)
+		.withLatestFrom(this.store$.select(toolsStateSelector))
+		.filter(([action, toolsState]: [any, IToolsState]) => toolsState.flags.get('shadowMouse'))
+		.map(() => new StartMouseShadow());
```
Comment (yairtawil) [inline src/app/app/app-effects/effects/map.app.effects.ts]: 💯

File: src/app/app/app-effects/effects/map.app.effects.ts [hunk 4]
```diff
@@ -555,6 +555,7 @@ export class MapAppEffects {
 	 * @action SetOverlayNotInCaseAction
 	 */
 	@Effect()
```
Comment (yairtawil) [inline src/app/app/app-effects/effects/map.app.effects.ts]: oops

File: src/app/app/app-effects/effects/map.app.effects.ts (hunk 1 above)
Comment (yuvalbl) [reply src/app/app/app-effects/effects/map.app.effects.ts]: Done.

File: src/app/app/app-effects/effects/map.app.effects.ts [hunk 5]
```diff
@@ -403,42 +375,50 @@ export class MapAppEffects {
 			return communicatorsIds.length > 1 && communicatorsIds.length === mapState.mapsList.length;
 		})
 		.mergeMap(([action, mapState, layerState]: [any, IMapState, ILayerState]) => {
-			const actionResults = [new CompositeMapShadowAction()];
+			const actions = new Array<Action>();
```
[the remark this answers is not in the document]
Comment (yuvalbl) [reply src/app/app/app-effects/effects/map.app.effects.ts]: Done.

File: src/app/app/app-effects/effects/map.app.effects.ts (hunk 4 above)
Comment (yuvalbl) [reply src/app/app/app-effects/effects/map.app.effects.ts]: ?

File: src/app/app/app-effects/effects/map.app.effects.spec.ts [hunk 6]
```diff
@@ -337,25 +336,21 @@ describe('MapAppEffects', () => {
 		expect(imagery1.removeVectorLayer).toHaveBeenCalledWith(staticLeaf);
 	});
 
-	describe('onCommunicatorChange$', () => {
+	describe('onCommunicatorChange$ ', () => {
```
Comment (yairtawil) [inline src/app/app/app-effects/effects/map.app.effects.spec.ts]: Amit

File: src/app/app/app-effects/effects/map.app.effects.spec.ts [hunk 7]
```diff
@@ -337,25 +336,21 @@ describe('MapAppEffects', () => {
...
+		actions = hot('--a--', {
+			a: new AddMapInstanceAction({
+				currentCommunicatorId: 'imagery2',
+				communicatorsIds: communicators
+			})
 		});
-
-		it('dont display annotation layer', () => {
+		const  b = new AnnotationVisualizerAgentAction({ maps: 'all', action: 'show' });
+		const expectedResults = cold('--b--', { b: b });
```
Comment (yairtawil) [inline src/app/app/app-effects/effects/map.app.effects.spec.ts]: Amit `{ b }`

File: src/app/app/app-effects/effects/map.app.effects.spec.ts [hunk 8]
```diff
@@ -337,24 +343,21 @@ describe('MapAppEffects', () => {
...
 				a: new AddMapInstanceAction({
 					currentCommunicatorId: 'imagery2',
 					communicatorsIds: communicators
 				})
 			});
-			const a = new CompositeMapShadowAction();
-			const b = new AnnotationVisualizerAgentAction({ maps: 'all', action: 'show' });
-			const expectedResults = cold('--(ab)--', { a, b });
+			const  b = new AnnotationVisualizerAgentAction({ maps: 'all', action: 'show' });
+			const expectedResults = cold('--b--', { b: b });
```
Comment (yairtawil) [inline src/app/app/app-effects/effects/map.app.effects.spec.ts]: Amit `{ b }`

**Pull request merged**